Client/Frontend/Browser/BrowserViewController/BrowserViewController+WebViewDelegates.swift

```diff
@@ -342,7 +350,16 @@ extension BrowserViewController: WKNavigationDelegate {
             decisionHandler(.cancel)
             return
         }
-
+        
+        if tab == tabManager.selectedTab, navigationAction.navigationType == .linkActivated, tab.adsTelemetryUrlList.count > 0 {
+            let adUrl = url.absoluteString
+            if tab.adsTelemetryUrlList.contains(adUrl) {
+                if !tab.adsProviderName.isEmpty { AdsTelemetryHelper.trackAdsClickedOnPage(providerName: tab.adsProviderName) }
+                tab.adsTelemetryUrlList.removeAll()
```

Comment:
@nbhasin2 The `adsTelemetryUrlList` and `adsProviderName` only get reset if the URL that is being navigated to is in the `adsTelemetryUrlList`. Is that correct? Should we not reset those two always after navigating to a new page?

Reply:
@st3fan You are right, in an ideal scenario we should reset `adsTelemetryUrlList` and `adsProviderName` on navigation. 

During my implementation I found that before we land on the final url there are many redirects that happen with search engine ads url before landing on final website and this was the best possible solution at the time to check if an ad url is part of redirects and once found we can then ignore other redirect urls. 

I am open to exploring any idea that you might have to cope with redirects and if Focus does it differently.

Reply:
No i think you are right about the redirects